tests/components/alert/test_init.py

```diff
@@ -117,239 +92,237 @@ def async_toggle(hass, entity_id):
+    assert STATE_IDLE == hass.states.get(ENTITY_ID).state
+    hass.states.async_set("sensor.test", STATE_ON)
+    await hass.async_block_till_done()
+    assert STATE_ON == hass.states.get(ENTITY_ID).state
+
+
+async def test_reset(hass, mock_notifier):
+    """Test resetting the alert."""
+    assert await async_setup_component(hass, alert.DOMAIN, TEST_CONFIG)
+    hass.states.async_set("sensor.test", STATE_ON)
+    await hass.async_block_till_done()
+    async_turn_off(hass, ENTITY_ID)
+    await hass.async_block_till_done()
+    assert STATE_OFF == hass.states.get(ENTITY_ID).state
+    async_turn_on(hass, ENTITY_ID)
+    await hass.async_block_till_done()
+    assert STATE_ON == hass.states.get(ENTITY_ID).state
+
+
+async def test_toggle(hass, mock_notifier):
+    """Test toggling alert."""
+    assert await async_setup_component(hass, alert.DOMAIN, TEST_CONFIG)
+    hass.states.async_set("sensor.test", STATE_ON)
+    await hass.async_block_till_done()
+    assert STATE_ON == hass.states.get(ENTITY_ID).state
+    async_toggle(hass, ENTITY_ID)
+    await hass.async_block_till_done()
+    assert STATE_OFF == hass.states.get(ENTITY_ID).state
+    async_toggle(hass, ENTITY_ID)
+    await hass.async_block_till_done()
+    assert STATE_ON == hass.states.get(ENTITY_ID).state
+
+
+async def test_notification_no_done_message(hass):
+    """Test notifications."""
+    events = []
+    config = deepcopy(TEST_CONFIG)
+    del config[alert.DOMAIN][NAME][alert.CONF_DONE_MESSAGE]
+
+    @callback
+    def record_event(event):
+        """Add recorded event to set."""
+        events.append(event)
+
+    hass.services.async_register(notify.DOMAIN, NOTIFIER, record_event)
+
+    assert await async_setup_component(hass, alert.DOMAIN, config)
+    assert len(events) == 0
+
+    hass.states.async_set("sensor.test", STATE_ON)
+    await hass.async_block_till_done()
+    assert len(events) == 1
+
+    hass.states.async_set("sensor.test", STATE_OFF)
+    await hass.async_block_till_done()
+    assert len(events) == 1
+
+
+async def test_notification(hass):
+    """Test notifications."""
+    events = []
+
+    @callback
+    def record_event(event):
+        """Add recorded event to set."""
+        events.append(event)
+
+    hass.services.async_register(notify.DOMAIN, NOTIFIER, record_event)
+
+    assert await async_setup_component(hass, alert.DOMAIN, TEST_CONFIG)
+    assert len(events) == 0
+
+    hass.states.async_set("sensor.test", STATE_ON)
+    await hass.async_block_till_done()
+    assert len(events) == 1
+
+    hass.states.async_set("sensor.test", STATE_OFF)
+    await hass.async_block_till_done()
+    assert len(events) == 2
+
+
+async def test_sending_non_templated_notification(hass, mock_notifier):
+    """Test notifications."""
+    assert await async_setup_component(hass, alert.DOMAIN, TEST_CONFIG)
+
+    hass.states.async_set(TEST_ENTITY, STATE_ON)
+    await hass.async_block_till_done()
+    assert len(mock_notifier) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)